src/main/java/com/denizenscript/depenizen/bukkit/commands/mythicmobs/MythicSkillCommand.java

```diff
@@ -3,113 +3,107 @@
-        }
-        ArrayList<Entity> entityTargets = null;
-        ArrayList<Location> locationTargets = null;
-        if (entity_targets != null) {
-            entityTargets = new ArrayList<>();
-            for (EntityTag entity : entity_targets) {
-                entityTargets.add(entity.getBukkitEntity());
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgLinear @ArgName("skill") ElementTag skill,
+                                   @ArgLinear @ArgName("targets") ListTag targets,
+                                   @ArgPrefixed @ArgName("power") @ArgDefaultText("1") ElementTag power,
+                                   @ArgPrefixed @ArgName("casters") @ArgDefaultNull ListTag casters,
+                                   @ArgPrefixed @ArgName("trigger") @ArgDefaultNull EntityTag trigger,
+                                   @ArgPrefixed @ArgName("origin") @ArgDefaultNull LocationTag origin,
+                                   @ArgPrefixed @ArgName("parameters") @ArgDefaultNull MapTag parameters,
+                                   @ArgPrefixed @ArgName("variables") @ArgDefaultNull MapTag variables,
+                                   @ArgLinear @ArgName("async") @ArgDefaultText("false") boolean async) {
+        List<Entity> entityTargets = new ArrayList<>();
+        List<Location> locationTargets = new ArrayList<>();
+        for (ObjectTag object : targets.objectForms) {
+            if (object.canBeType(EntityTag.class)) {
+                entityTargets.add(((EntityTag) object).getBukkitEntity());
+            }
+            else if (object.canBeType(LocationTag.class)) {
+                locationTargets.add((Location) object);
             }
         }
-        else {
-            locationTargets = new ArrayList<>(location_targets);
+        if (!entityTargets.isEmpty() && !locationTargets.isEmpty()) {
+            Debug.echoError("Cannot have both entity and location targets.");
+            return;
         }
-        for (EntityTag caster : casters) {
-            MythicMobsBridge.getAPI().castSkill(caster.getBukkitEntity(), skill.asString(), caster.getBukkitEntity().getLocation(), entityTargets, locationTargets, power.asFloat());
+
+        for (EntityTag caster : casters != null ? casters.filter(EntityTag.class, scriptEntry) : Utilities.entryDefaultEntityList(scriptEntry, true)) {
+            MythicMobsBridge.getAPI().castSkill(caster.getBukkitEntity(), skill.asString(), trigger != null ? trigger.entity : caster.getBukkitEntity(), origin, entityTargets, locationTargets, power.asFloat() , (metadata) -> {
+                if (parameters != null) {
+                    Map<String, String> parameterMap = new HashMap<>();
+                    for (Map.Entry<StringHolder, ObjectTag> entry : parameters.map.entrySet()) {
+                        parameterMap.put(entry.getKey().toString(), entry.getValue().toString());
```

Comment:
`entry.getKey().str`

Reply:
are parameters case-sensitive? If so... yike. If not, should use `.low` or an appropriate casing standardization method

Reply:
yeah they are